### DISCUSSION TITLE
Default  value of the Pod CIDR is 10.42.0.0/24，I want to customize it.

### DISCUSSION THREAD

**Discussion by jack331200** [k3s-io/k3s#7235] [Q&A] [answered]
https://github.com/k3s-io/k3s/issues/697#issuecomment-576463182
After reviewing the history, I found that some parameters could solve my problem. However, I want to customize the Pod CIDR on different node nodes.

```
k3s]# cat /run/flannel/subnet.env 
FLANNEL_NETWORK=10.30.0.0/16
FLANNEL_SUBNET=10.30.0.1/24
FLANNEL_MTU=1450
FLANNEL_IPMASQ=true
```
Customized
```
k3s]# cat /run/flannel/subnet.env 
FLANNEL_NETWORK=10.30.0.0/16
FLANNEL_SUBNET=10.30.21.1/24
FLANNEL_MTU=1450
FLANNEL_IPMASQ=true
```

`FLANNEL_SUBNET` 
The Pod CIDR needs to be defined during deployment, but I couldn't find a solution in my research materials. I hope to receive help.

**Comment by brandond** [comment, marked as answer]
https://docs.k3s.io/cli/server#networking
> Flag | Default | Description
> -- | -- | --
> --cluster-cidr value | "10.42.0.0/16" | IPv4/IPv6 network CIDRs to use for pod IPs
> --service-cidr value | "10.43.0.0/16" | IPv4/IPv6 network CIDRs to use for service IPs

You cannot manually set the pod CIDR on individual nodes using the cluster-cidr flag, all you can do is set the cluster cidr and node-cidr-mask.

There is an **experimental** Kubernetes feature that will let you do what you want, called Multi Cluster-CIDR. You can find docs for this at https://docs.k3s.io/networking#multicluster-cidr-experimental

**Reply by jack331200** [reply]
Thank you, I'll test it out.
